File: 1968.array-with-elements-not-equal-to-average-of-neighbors.cpp
/*
 * @lc app=leetcode id=1968 lang=cpp
 *
 * [1968] Array With Elements Not Equal to Average of Neighbors
 */

// @lc code=start
#include <vector>
#include <algorithm>

class Solution {
public:
    std::vector<int> rearrangeArray(std::vector<int>& nums) {
        int n = nums.size();
        std::sort(nums.begin(), nums.end());
        std::vector<int> result(n, 0);
        for (int i = 0; i < n; i += 2) {
            result[i] = nums[n - 1 - i / 2];
        }
        for (int i = 1; i < n; i += 2) {
            result[i] = nums[i / 2];
        }
        return result;
    }
};

// @lc code=end

File: 1968.array-with-elements-not-equal-to-average-of-neighbors.py
#
# @lc app=leetcode id=1968 lang=python3
#
# [1968] Array With Elements Not Equal to Average of Neighbors
#

# @lc code=start
from typing import List


class Solution:
    '''
    We can sort nums first
    Then put the large ones on odd and small ones on even indices
    '''
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        nums.sort()
        result = [0] * n
        for i in range(0, n, 2):
            result[i] = nums[n - 1 - i // 2]
        for i in range(1, n, 2):
            result[i] = nums[i // 2]
        return result
# @lc code=end

